crates/bevy_feathers/src/cursor.rs

```diff
@@ -0,0 +1,69 @@
+use bevy_app::{App, Plugin, PreUpdate};
+use bevy_ecs::{
+    entity::Entity,
+    hierarchy::ChildOf,
+    resource::Resource,
+    schedule::IntoScheduleConfigs,
+    system::{Commands, Query, Res},
+};
+use bevy_picking::{hover::HoverMap, pointer::PointerId, PickingSystems};
+use bevy_window::Window;
+use bevy_winit::cursor::CursorIcon;
+
+/// A component that specifies the cursor icon to be used when the mouse is not hovering over
+/// any other entity. This is used to set the default cursor icon for the window.
+#[derive(Resource, Debug, Clone, Default)]
+pub struct DefaultCursorIcon(pub CursorIcon);
+
+/// System which updates the window cursor icon whenever the mouse hovers over an entity with
+/// a [`CursorIcon`] component. If no entity is hovered, the cursor icon is set to
```

Comment:
Follow-up: we should be updating the docs for `bevy::window::CursorIcon` to signal to users that this is an intended pattern, and moving this code out into `bevy_picking` or something.